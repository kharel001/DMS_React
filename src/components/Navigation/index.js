import React from 'react';
import {List} from '@material-ui/core';
import NavMenuItem from "./NavMenuItem";
import NavSection from "./NavSection";
import NavCollapse from "./NavCollapse";

const Navigation = props => {
  const {menuItems} = props;
  return (
    <List component="nav" disablePadding className='side-nav-menu'>
      {
        menuItems.map((item, index) => {
          switch (item.type) {
            case 'section' :
              return <NavSection {...item} key={index} data-test="sectionComp"/>;
            case 'collapse' :
              return <NavCollapse {...item} key={index} data-test="collapseComp"/>;
            case 'item' :
              return <NavMenuItem {...item} key={index} data-test="itemComp"/>;
          }
        })
      }
    </List>
  )
};

export default Navigation;