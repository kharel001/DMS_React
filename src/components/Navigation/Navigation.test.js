import React from 'react';
import Navigation from './index';
import {
    configure,
    shallow,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})
const dummy_data= {
  menuItems:[{id:1, item:"item1", type:"section"},
  {id:2, item:"item2", type:"collapse"},
  {id:3, item:"item3",type:"item"}]
}
const setup = (props = {}) => {
    return shallow( < Navigation
     {...props}  />
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}

{/* it("Custom Scrollable render", () => {
        let wrapper = setup(dummy_data);
        // expect(wrapper).toMatchSnapshot();
}) */}
it("Snapshot Navigation", () => {
    let wrapper = setup(dummy_data);
    expect(wrapper).toMatchSnapshot();
})



it("Components Navigation", () => {
    let wrapper = setup(dummy_data);
    let sectionComp = findByTestAttr(wrapper, "sectionComp");
    expect(sectionComp).toHaveLength(1);
    let collapseComp = findByTestAttr(wrapper, "collapseComp");
    expect(collapseComp).toHaveLength(1);
        let itemComp = findByTestAttr(wrapper, "itemComp");
    expect(itemComp).toHaveLength(1);
   
})