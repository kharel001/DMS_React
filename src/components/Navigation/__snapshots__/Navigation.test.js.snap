// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Navigation 1`] = `ShallowWrapper {}`;
